astroquery/solarsystem/pds/core.py

```diff
@@ -0,0 +1,368 @@
+            of length).  If ``None``, then the geocenter is used.
+        neptune_arcmodel : float, optional. which ephemeris to assume for Neptune's ring arcs
+            must be one of 1, 2, or 3 (see https://pds-rings.seti.org/tools/viewer3_nep.shtml for details)
+            has no effect if planet != 'Neptune'
+        get_query_payload : boolean, optional
+            When set to `True` the method returns the HTTP request parameters as
+            a dict, default: False
+
+
+        Returns
+        -------
+        response : `requests.Response`
+            The response of the HTTP request.
+
+
+        Examples
+        --------
+        >>> from astroquery.solarsystem.pds import RingNode
+        >>> import astropy.units as u
+        >>> systemtable, bodytable, ringtable = RingNode.ephemeris(planet='Uranus',
+        ...                 epoch='2024-05-08 22:39',
+        ...                 location = (-23.029 * u.deg, -67.755 * u.deg, 5000 * u.m))  # doctest: +REMOTE_DATA
+        >>> print(ringtable)  # doctest: +REMOTE_DATA
+              ring  pericenter ascending node
+                       deg          deg
+            ------- ---------- --------------
+                Six    293.129           52.0
+               Five    109.438           81.1
+               Four    242.882           66.9
+              Alpha    184.498          253.9
+               Beta     287.66          299.2
+                Eta        0.0            0.0
+              Gamma     50.224            0.0
+              Delta        0.0            0.0
+             Lambda        0.0            0.0
+            Epsilon    298.022            0.0
+        """
+
+        planet = planet.lower()
+        if planet not in ["mars", "jupiter", "saturn", "uranus", "neptune", "pluto", ]:
+            raise ValueError(
+                "illegal value for 'planet' parameter (must be 'Mars', 'Jupiter', 'Saturn', 'Uranus', 'Neptune', or 'Pluto')"
+            )
+
+        if isinstance(epoch, (int, float)):
+            epoch = Time(epoch, format='jd')
+        elif isinstance(epoch, str):
+            epoch = Time(epoch, format='iso')
+        elif epoch is None:
+            epoch = Time.now()
+            log.warning("Observation time not set. Using current time.")
+
+        if location is None:
+            viewpoint = "observatory"
+            latitude, longitude, altitude = "", "", ""
+            log.warning("Observatory coordinates not set. Using center of Earth.")
+        else:
+            viewpoint = "latlon"
+            if isinstance(location, EarthLocation):
+                loc = location.geodetic
+                longitude = loc[0].deg
+                latitude = loc[1].deg
+                altitude = loc[2].to(u.m).value
+            elif hasattr(location, "__iter__"):
+                latitude = Angle(location[0]).deg
+                longitude = Angle(location[1]).deg
+                altitude = u.Quantity(location[2]).to("m").value
```

Comment:
I would find it less confusing if your code would expect longitude and latitude in the same order as `EarthLocation`.